lib/aws.js

```diff
@@ -10,15 +10,17 @@ const logger = require('./logger');
 const config = require('./config');
 
 module.exports.init = function(callback) {
-    // AWS will look relative to the Node working directory, not the current
-    // directory. So aws-config.json should be in the project root.
-    //
-    // If it exists, load it. If it does not exist, then we are either (1)
-    // running locally, and so probably don't care, or (2) running on prod,
+    // If we are runningInEc2, then we are running on prod,
     // and so AWS will have already been configured.
-    if (fs.existsSync('./aws-config.json')) {
-        logger.info('Loading AWS credentials for external grading and workspace S3 resources');
-        AWS.config.loadFromPath('./aws-config.json');
+    if (!config.runningInEc2) {
+        const configFile = path.join(process.cwd(), 'aws-config.json');
+        if (fs.existsSync(configFile)) {
```

Comment:
Was this `*Sync` call intentional or just old code? I read that sometimes it's good to load configs using `*Sync`? The `lib/workspace` version used async.

Reply:
I think async would be better. We correctly sequence all the initialization code when we call it using `async.series()`, so async will work fine and I think it's cleaner to always use async.

Reply:
Ok, `lib/aws.init()` should be async now: d275290

Reply:
Excellent!